xdsl/dialects/mpi.py

```diff
@@ -78,6 +80,23 @@ class DataType(ParametrizedAttribute, MLIRType):
     name = 'mpi.datatype'
 
 
+VectorWrappable = RequestType | StatusType | DataType
+_VectorT = TypeVar('_VectorT', bound=VectorWrappable)
+
+
+@irdl_attr_definition
+class VectorType(Generic[_VectorT], ParametrizedAttribute, MLIRType):
```

Comment:
Why is the builtin vectortype not enough here?

Reply:
I'm not 100% sure, I think I looked at the docs for `vector` and was put off by the fact that it was specifically about stuff that should be vectorized.

We can def. move that to the builtin `vector` later on!